src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/HearingDocumentService.java

```diff
@@ -132,4 +147,82 @@ public void sendToBulkPrint(CaseDetails caseDetails, String authorisationToken)
+            .courtName((String) courtDetails.get(COURT_DETAILS_NAME_KEY))
+            .courtAddress((String) courtDetails.get(COURT_DETAILS_ADDRESS_KEY))
+            .phoneNumber((String) courtDetails.get(COURT_DETAILS_PHONE_KEY))
+            .email((String) courtDetails.get(COURT_DETAILS_EMAIL_KEY))
+            .build();
+
+        caseData.put("ccdCaseNumber", caseDetails.getId());
+        caseData.put("DivorceCaseNumber", caseDetails.getData().get(DIVORCE_CASE_NUMBER));
+        caseData.put("ApplicantName", buildFullApplicantName(caseDetails));
+        caseData.put("RespondentName", buildFullRespondentName(caseDetails));
+        caseData.put("HearingType", caseDetails.getData().get(HEARING_TYPE));
+        caseData.put("HearingVenue", selectedFRCDetails.getCourtName());
+        caseData.put("HearingDate", HEARING_DATE);
+        caseData.put("HearingTime", HEARING_TIME);
+        caseData.put("HearingLength", TIME_ESTIMATE);
+        caseData.put("AnyOtherDirections", HEARING_ADDITIONAL_INFO);
+
+        caseData.put("CourtName", selectedFRCDetails.getCourtName());
+        caseData.put("CourtAddress", selectedFRCDetails.getCourtAddress());
+        caseData.put("CourtPhone", selectedFRCDetails.getPhoneNumber());
+        caseData.put("CourtEmail", selectedFRCDetails.getEmail());
+    }
+
+    private void addAdditionalHearingDocumentToCaseData(CaseDetails caseDetails, CaseDocument document) {
+        AdditionalHearingDocumentData generatedDocumentData = AdditionalHearingDocumentData.builder()
+            .additionalHearingDocument(AdditionalHearingDocument.builder()
+                .additionalHearingDocument(document)
+                .build())
+            .build();
+
+        Map<String, Object> caseData = caseDetails.getData();
+        List<AdditionalHearingDocumentData> additionalHearingDocumentDataList =
+            Optional.ofNullable(caseData.get(ADDITIONAL_HEARING_DOCUMENT_COLLECTION))
+                .map(documentHelper::convertToAdditionalHearingDocumentData)
+                .orElse(new ArrayList<>(1));
+
+        additionalHearingDocumentDataList.add(generatedDocumentData);
+
+        caseData.put(ADDITIONAL_HEARING_DOCUMENT_COLLECTION, additionalHearingDocumentDataList);
+    }
+
+    private void bulkPrintAdditionalHearingDocuments(CaseDetails caseDetails, String authorisationToken) {
+        List<AdditionalHearingDocumentData> additionalHearingDocumentData =
+            documentHelper.convertToAdditionalHearingDocumentData(
+                caseDetails.getData().get(ADDITIONAL_HEARING_DOCUMENT_COLLECTION));
+
+        AdditionalHearingDocumentData additionalHearingDocument = additionalHearingDocumentData.get(additionalHearingDocumentData.size() - 1);
+
+        List<BulkPrintDocument> document = Collections.singletonList(
+            bulkPrintService.getBulkPrintDocumentFromCaseDocument(
+                additionalHearingDocument.getAdditionalHearingDocument().getAdditionalHearingDocument()));
```

Comment:
this looks like we're using complex types where we shouldn't be. we should really only have to do AdditionalHearingDocumentData. getAdditionalHearingDocument() here

Reply:
I think this is a poor naming convention on my part. It is now:
additionalHearingDocument.getAdditionalHearingDocument().getDocument()

If you check AdditionalHearingDocumentData.java, the variable is called additionalHearingDocument and in the AdditionalHearingDocument.java it is the same name! I will updated this in my next push